symfony/plugins/orangehrmLeavePlugin/Service/LeaveAssignmentService.php

```diff
@@ -19,101 +19,120 @@
     }
 
-    public function getDispatcher() {
-        if(is_null($this->dispatcher)) {
+    public function getDispatcher()
+    {
+        // TODO
+        if (is_null($this->dispatcher)) {
             $this->dispatcher = sfContext::getInstance()->getEventDispatcher();
         }
         return $this->dispatcher;
-    }  
-    
+    }
+
     /**
-     *
-     * @param array $leaveAssignmentData
-     * @return bool
+     * @param LeaveParameterObject $leaveAssignmentData
+     * @return LeaveRequest|null
+     * @throws LeaveAllocationServiceException
      */
-    public function assignLeave(LeaveParameterObject $leaveAssignmentData) {
-
+    public function assignLeave(LeaveParameterObject $leaveAssignmentData)
+    {
         $employeeId = $leaveAssignmentData->getEmployeeNumber();
-
         /* Check whether employee exists */
         if (empty($employeeId)) {
-            throw new LeaveAllocationServiceException('Invalid Employee');
+            throw LeaveAllocationServiceException::invalidEmployeeSelected();
         }
-
         if ($this->hasOverlapLeaves($leaveAssignmentData)) {
-            return false;
+            throw LeaveAllocationServiceException::overlappingLeavesFound();
         }
-        
         if ($this->isWorkShiftLengthExceeded($leaveAssignmentData)) {
-            throw new LeaveAllocationServiceException('Failed to Assign: Work Shift Length Exceeded');
+            throw LeaveAllocationServiceException::workShiftLengthExceeded();
         }
-                
+
         return $this->saveLeaveRequest($leaveAssignmentData);
     }
 
     /**
      * Saves Leave Request and Sends Notification
-     * 
-     * @param LeaveParameterObject $leaveAssignmentData 
-     * 
+     *
+     * @param LeaveParameterObject $leaveAssignmentData
+     * @return LeaveRequest|null True if leave request is saved else false
+     * @throws LeaveAllocationServiceException
      */
-    protected function saveLeaveRequest(LeaveParameterObject $leaveAssignmentData) {
-
+    protected function saveLeaveRequest(LeaveParameterObject $leaveAssignmentData): ?LeaveRequest
+    {
         $leaveRequest = $this->generateLeaveRequest($leaveAssignmentData);
-
-          $leaveType = $this->getLeaveTypeService()->readLeaveType($leaveAssignmentData->getLeaveType());
+        $leaveType = $this->getLeaveTypeService()->getLeaveTypeDao()->getLeaveTypeById(
+            $leaveAssignmentData->getLeaveType()
+        );
 //        $leaveRequest->setLeaveTypeName($leaveType->getLeaveTypeName());
-
 //        if (is_null($leaveRequest->getLeavePeriodId())) {
 //            if ($this->getLeavePeriodService()->isWithinNextLeavePeriod(strtotime($leaveRequest->getDateApplied()))) {
 //                $nextLeavePeriod = $this->getLeavePeriodService()->createNextLeavePeriod($leaveRequest->getDateApplied());
 //                $leaveRequest->setLeavePeriodId($nextLeavePeriod->getLeavePeriodId());
 //            }
 //        }
```

Comment:
remove this commented code, seems old one

Reply:
Done